### PR TITLE
docs: icons component

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

- Added icon component documentation.
- Removed old icons documentation and added a link to version6 in case its needed.
- Added a category filter
- Sorted the icon array
- Added @dialpad/dialtone-icons

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

**Review by francisrupert — COMMENTED** [review]
@juliodialpad Absolutely frickin awesome work. I've pushed a few quick changes, primarily styles before aligning to the Figma prototype for searching icons (especially redoing the popover).

Screenshot of before/after of those style tweaks made in https://github.com/dialpad/dialtone/pull/731/commits/3032488f78df2161eac00a88f652f88309947b4f

image

I'll hold off on approval to make sure @braddialpad and/or @josedialpad get a chance to review the Vue approach.

**Review by francisrupert — CHANGES_REQUESTED** [review]
We'll need to retain the **Illustrations** section. 

image

**Review by francisrupert — CHANGES_REQUESTED** [review]
Let's also retain the previous **Icons** section in the meantime too...

image

...and continue to use the Notice to point to the new icons as currently done at https://dialpad.design/design/icons/brand.html, https://dialpad.design/design/icons/system.html, https://dialpad.design/design/icons/weather.html, https://dialpad.design/design/icons/sizes.html...

image

And revert to this notice on https://dialpad.design/components/icon.html.

I'll be reorganizing the https://dialpad.design/design section to follow the Landing Page pattern I've begun with Components landing page, and thus better speak to previous icons vs Icons 7.

**Review by francisrupert — APPROVED** [review]

**Comment by francisrupert** [comment]
Sweet, let's move this on!

File: docs/.vuepress/client.js [hunk 1]
```diff
@@ -65,45 +65,45 @@ export default defineClientConfig({
...
+//   });
+//
+//   await Promise.all(iconPromises);
+//
+//   iconEntries.forEach(([name, icon]) => {
+//     app.component(name, icon);
+//   });
+//
+//   app.provide('dialtoneIcons', iconEntries.map(([name]) => name));
+// }
```
Comment (braddialpad) [inline docs/.vuepress/client.js]: Is this breaking something? I'm actually using the dialtone vue components on a branch currently.

**Review by braddialpad — APPROVED** [review]
Looks great thanks!

File: docs/.vuepress/client.js (hunk 1 above)
Comment (juliodialpad) [reply docs/.vuepress/client.js]: Not really, was getting a lot of warnings of "Already registered component"

Thought it wasn't being used, I'll uncomment and merge

**Review by francisrupert — COMMENTED** [review]
Getting there! I provisionally approve, though the main diff w/ this last commit is Vue-specific so I'll defer to @braddialpad for full-on PR approval.

**Comment by francisrupert** [comment]
heck to the yeah

**Pull request merged**